### ISSUE TITLE
Overlays wobble on drag (sub-pixel offsets rounding problem)

### ISSUE THREAD

**Issue by EvilTosha** [openseadragon/openseadragon#1474]
I've encountered an issue with overlays in Chrome on Mac. Actually, it can be observed on the site: https://openseadragon.github.io/examples/ui-overlays/ - try zooming in and then drag the image; the blue frame will wobble 1px to all directions.

The problem is that browsers (at least chrome) don't handle sub-pixel offsets (top, left, etc) and end up rounding them to the nearest integer. However, they do handle sub-pixel "transform:translate" well. I've tested the hypothesis by setting translate instead of top/left offsets here, and it fixed the issue.

If somebody with good expertise with OSD code could implement this change (taking into account possible rotations), that would be amazing.

**Comment by iangilman** [comment]
Good point! Now that we no longer support IE8, we can switch to using transform for this. Hopefully someone will take it on.

**Comment by iangilman** [comment]
@msalsbery looks like you haven't done anything on this… Should we unassign you?

**Comment by msalsbery** [comment]
@iangilman yes, done! I have a large queue of assigned MouseTracker issues I’m into currently so it would be a while before I got to this. Thanks!

**Comment by iangilman** [comment]
Cool. Hopefully someone can take this on :-)

**Comment by Medwar** [comment]
Thank you guys, i've fixed it, works like a charm. 
I will have to test rotation and cross-browserness as well.

I am going to contribute this later after I test it for a while.

line 17283:
`        drawHTML: function(container, viewport) {
            var element = this.element;
            if (element.parentNode !== container) {
                //save the source parent for later if we need it
                element.prevElementParent = element.parentNode;
                element.prevNextSibling = element.nextSibling;
                container.appendChild(element);

                // have to set position before calculating size, fix #1116
                this.style.position = "absolute";
                // this.size is used by overlays which don't get scaled in at
                // least one direction when this.checkResize is set to false.
                this.size = $.getElementSize(element);
            }

            var positionAndSize = this._getOverlayPositionAndSize(viewport);

            var position = positionAndSize.position;
            var size = this.size = positionAndSize.size;
            var rotate = positionAndSize.rotate;

            // call the onDraw callback if it exists to allow one to overwrite
            // the drawing/positioning/sizing of the overlay
            if (this.onDraw) {
                this.onDraw(position, size, this.element);
            } else {
                var style = this.style;
                var transformOriginProp = $.getCssPropertyWithVendorPrefix(
                    'transformOrigin');
                var transformProp = $.getCssPropertyWithVendorPrefix(
                    'transform');                
                style[transformProp] = "translate("+position.x+"px,"+position.y+"px)";
                // style.left = position.x + "px";     // maybe leave this with condition - if transform is not supported?
                // style.top = position.y + "px";     // maybe leave this if transform is not supported?
                if (this.width !== null) {
                    style.width = size.x + "px";
                }
                if (this.height !== null) {
                    style.height = size.y + "px";
                }
                if (transformOriginProp && transformProp) {
                    if (rotate) {
                        style[transformOriginProp] = this._getTransformOrigin();
                        style[transformProp] = style[transformProp] + "rotate(" + rotate + "deg)";
                    } else {
                        style[transformOriginProp] = "";
                        //style[transformProp] = "";    
                    }
                }

                if (style.display !== 'none') {
                    style.display = 'block';
                }
            }
        },   `

**Comment by iangilman** [comment]
Awesome! Thank you for doing this! I look forward to the pull request once you're happy with the testing results.

**Comment by LeoSeyers** [comment]
Thanks @Medwar you're my today's savior. I was running into poor performance issues on mobile and was highly suspecting the overlays to cause the bottlenecks. I had about 50 of theses. After inspection, animating through top and left properties is verly likely the issue. The fix improved performance a lot. This should defintely be implemented into the librairy :)

**Comment by iangilman** [comment]
@LeoSeyers Great to hear that helped! Looks like @Medwar never got around to making the PR. If you're up for pitching in, I would welcome a PR from you. Otherwise, hopefully someone will take an interest :)

**Comment by MatthiasDurivet** [comment]
I've also noticed this "wobble", even sites like Excalidraw show this wobble behavior.
For me it's occurring when creating a new rectangle, by press&dragging from one point to another point.

One way to solve it is to make sure the moving (sub-pixel) corner is on the side of the cursor. The opposite corner will not encounter this "wobble". This happens because our brains only notice the wobble on the corner that should be stationary but our brains don't notice the wobble on the moving corner.
